### PR TITLE
[DQL-65] Automated tests

### PR DESCRIPTION
**Pull request by salsa-nathan**
[DQL-65]

This PR:
- fixes a code formatting issue in `plugin.py`
- introduces the `navigation.feature` with BDD tests for the header and footer.

### CODE REVIEW

File: test/features/homepage.feature [hunk 1]
```diff
@@ -1,6 +0,0 @@
-@smoke
```
Comment (alexdesignworks) [inline test/features/homepage.feature]: as in other repos - please revert this . it is required to make sure that all systems are working correctly

File: test/features/login.feature [hunk 2]
```diff
@@ -1,18 +0,0 @@
-@smoke
-Feature: Login
```
Comment (alexdesignworks) [inline test/features/login.feature]: as above

File: test/features/navigation.feature [hunk 3]
```diff
@@ -0,0 +1,23 @@
+@smoke
```
Comment (alexdesignworks) [inline test/features/navigation.feature]: please remove `@smoke` tag

File: test/features/navigation.feature [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+Feature: Navigation
+
+    Scenario: DQL-65 BDD-1 "Request data" link in header (non-logged in user)
```
Comment (alexdesignworks) [inline test/features/navigation.feature]: please remove `DQL-*` prefix

File: test/features/steps/steps.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 from behaving.web.steps.url import when_i_visit_url
 
 
-@when('I go to homepage')
+@step('I go to homepage')
```
Comment (alexdesignworks) [inline test/features/steps/steps.py]: 💯

File: test/features/homepage.feature (hunk 1 above)
Comment (salsa-nathan) [reply test/features/homepage.feature]: homepage.feature smoke test re-instated.

File: test/features/login.feature (hunk 2 above)
Comment (salsa-nathan) [reply test/features/login.feature]: login.feature smoke tests re-instated.

File: test/features/navigation.feature (hunk 4 above)
Comment (salsa-nathan) [reply test/features/navigation.feature]: Jira story references removed from Scenario descriptions and Scenarios elaborated as per William's request on other BDD PRs.

File: test/features/navigation.feature (hunk 3 above)
Comment (salsa-nathan) [reply test/features/navigation.feature]: @smoke tag removed.

File: test/features/navigation.feature [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    Scenario: Check for the presence of the "Request data" link in header when visiting as a logged in user
+        Given "Admin" as the persona
+        When I log in
+        Then I go to homepage
+        # Make the comparison case-insensitive
+        Then I should see an element with xpath "//a[contains(translate(., 'RD', 'rd'), "request data")]"
+
+    Scenario: Check for the presence of the "Copyright" link in footer when visiting as a non-logged in user
+        When I go to homepage
```
Comment (duttonw) [inline test/features/navigation.feature]: Why no given?

as all of these tests start at the homepage? 

use background: to prep the system to be on the home page.
https://sukesh15.gitbooks.io/cucumber-jvm-test-framework-/content/cucumber_-_more_details/understanding_background_in_cucumber.html

**Pull request merged**